pkg/graveler/settings/manager.go

```diff
@@ -160,6 +160,7 @@ func (m *KVManager) Update(ctx context.Context, repositoryID graveler.Repository
 		}
 		err = m.store.SetMsgIf(ctx, graveler.RepoPartition(&graveler.RepositoryRecord{RepositoryID: repositoryID, Repository: repo}), []byte(graveler.SettingsPath(key)), data, pred)
 		if errors.Is(err, kv.ErrPredicateFailed) {
+			logging.Default().WithError(err).Warn("Predicate failed on settings update. Retrying")
```

Comment:
Warning is preferred and not an error, since this is expected for some usages of the settings manager.